frontend/src/Frontend/VersionedStore/V2.hs

```diff
@@ -153,32 +153,37 @@ fromMultiSet = ($ []) . Map.foldrWithKey (\k i -> (.) (dlrep k i)) id
       | n == 0 = id
       | otherwise = (v:) . dlrep v (n - 1)
 
-data Cheese a b = That b | These a b
-
-instance Functor (Cheese a) where
-  fmap f (That b) = That (f b)
-  fmap f (These a b) = These a (f b)
-
-instance (Semigroup a) => Applicative (Cheese a) where
-  pure = That
-  That f <*> That x = That (f x)
-  That f <*> These b x = These b (f x)
-  These a f <*> That x = These a (f x)
-  These a f <*> These b x = These (a <> b) (f x)
+-- It is equivalent to "These () a" except that we take out the
+-- "This" constructor. With it gone, we still get the same applicative
+-- operations from "These" (sans "This") but we don't have to account for
+-- the "This" constructor in situations where we know it is impossible to
+-- produce it. This is also morally equivalent to Either () b with a
+-- different applicative instance
+data Deez b = Dis b | Dat b
```

Comment:
slightly misleading comment. 

```haskell
These a b 
~ a + b + a*b
```

But

```haskell
These () a 
~ 1 + a + 1*a 
~ 1 + a + a 
~ Maybe (Either a a)
```

and 

```haskell
Either () b 
~ 1 + b
~ Maybe b
```

You want `Either a a` here. If you want `These` sans the `This` case, then you want `Either a (a,a)`